docs/environments/sokoban.md

```diff
@@ -0,0 +1,76 @@
+# Boxoban Environment 👾
+
+<p align="center">
+        <img src="../env_anim/sokoban.gif" width="500"/>
+</p>
+
+This is a Jax implementation of the _Sokoban_ puzzle, a dynamic box-pushing environment where the agent's goal is to place all boxes on their targets. This version follows the rules from the DeepMind paper on [Imagination Augmented Agents](https://arxiv.org/abs/1707.06203), with levels based on the Boxoban dataset from [Guez et al., 2018](https://github.com/deepmind/boxoban-levels)[[1]](#ref1). The graphical assets were taken from [gym-sokoban](https://github.com/mpSchrader/gym-sokoban) by Schrader, a diverse Sokoban library implementing many versions of the game in the OpenAI gym framework [[2]](#ref2).
+
+## Observation
+
+- `grid`: An Array (uint8) of shape `(10, 10, 2)`. It represents the variable grid (containing movable objects: boxes and the agent) and the fixed grid (containing fixed objects: walls and targets).
+- `step_count`: An Array (int32) of shape `()`, representing the current number of steps in the episode.
+
+## Object Encodings
+
+| Object       | Encoding |
+|--------------|----------|
+| Empty Space  | 0        |
+| Wall         | 1        |
+| Target       | 2        |
+| Agent        | 3        |
+| Box          | 4        |
+
+## Actions
+
+The agent's action space is an Array (int32) with potential values of `[0,1,2,3]` (corresponding to `[Up, Down, Left, Right]`). If the agent attempts to move into a wall, off the grid, or push a box into a wall or off the grid, the grid state remains unchanged; however, the step count is incremented by one. Chained box pushes are not allowed and will result in no action.
```

Comment:
Could we please change the order of actions in the action space for consistency? In all the related environments, we have `[0,1,2,3] -> [Up, Right, Down, Left]`.